src/main/java/org/fogbowcloud/arrebol/execution/DockerTaskExecutor.java

```diff
@@ -1,146 +1,200 @@
         return taskExecutionResult;
     }
 
-    private Integer start() {
+    protected Integer start(){
         try {
-            String[] cmd = {
-                    BASH,
-                    "-c",
-                    DOCKER_RUN + " " + this.containerName + " " + this.imageId
-            };
-            Process p = Runtime.getRuntime().exec(cmd);
-
-            Integer exitCode = p.waitFor();
-            return exitCode;
-        } catch (Exception e){
-            throw new RuntimeException("Error while trying execute commands to start container " + this.containerName);
+            LOGGER.info("Starting DockerTaskExecutor " + this.getContainerName());
+            this.workerDockerRequestHelper.start();
+            return new Integer(0);
+        } catch (Exception e) {
+            return new Integer(127);
         }
     }
 
-    private Integer stop() {
+    protected Integer stop(){
         try {
-            String[] cmd = {
-                    BASH,
-                    "-c",
-                    DOCKER_STOP + " " + this.containerName
-            };
-            Process p = Runtime.getRuntime().exec(cmd);
-
-            Integer exitCode = p.waitFor();
-            return exitCode;
-
-        } catch (Exception e){
-            throw new RuntimeException("Error while trying execute commands to stop container " + this.containerName);
+            LOGGER.info("Stopping DockerTaskExecutor " + this.getContainerName());
+            this.workerDockerRequestHelper.stop();
+            return new Integer(0);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new Integer(127);
         }
     }
 
-    private Integer executeCommand(Command command) {
-        try {
-            String commandStr = command.getCommand();
-            String[] cmd = {
-                    BASH,
-                    "-c",
-                    DOCKER_EXEC + " " + this.containerName + " " + commandStr
-            };
-            ProcessBuilder builder = new ProcessBuilder(cmd);
-            Process process = builder.start();
-            Integer exitCode = process.waitFor();
-            return exitCode;
-
-        } catch(Exception e){
-            throw new RuntimeException("Error while truing execute commands to container " + this.containerName);
+    protected Integer executeCommand(Command command) throws Exception {
+        LOGGER.info("Executing command [" + command.getCommand() + "] in worker [" + this.getContainerName() + "].");
+        String execId = this.workerDockerRequestHelper.createExecInstance(command.getCommand());
+        this.workerDockerRequestHelper.startExecInstance(execId);
+        ExecInstanceResult execInstanceResult = this.workerDockerRequestHelper.inspectExecInstance(execId);
+        while(execInstanceResult.getExitCode() == null){
+            execInstanceResult = this.workerDockerRequestHelper.inspectExecInstance(execId);
+            try {
+                sleep(300);
```

Comment:
what is `300`? we need a variable to it.

also, we need to doc it. I know we are pooling the result but in the future, we should no streamline execution.
a trade-off solution now or to the near future (it is best) is to have the pooling period in an optional config